### PR TITLE
Expose DeepSpeed FP16 parameters due to loss instability

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Received a few reports that DeepSpeed quickly gives off NaNs when using ZeRO. This seems to be related to default Loss scaling values, which are not exposed currently. As a result the user needs to override everything to set these values.

Expose these values so that the user can access them. I thought about putting the values into the Precision plugin, but then I'd need the training_type_plugin to be aware of the precision plugin and I'm not a fan of that, given a longer term plan of the Training Type plugin handling precision.

Also I've included a few tests of edge cases that needed to be checked + seeing if we need to run the single GPU tests as special tests (might need to revert this, we'll see).

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/plugins/test_deepspeed_plugin.py [hunk 1]
```diff
@@ -304,6 +295,58 @@ def on_train_start(self) -> None:
...
+            assert self.trainer.training_type_plugin.config['fp16']['loss_scale_window'] == 10
+            assert self.trainer.training_type_plugin.config['fp16']['hysteresis'] == 10
+            assert self.trainer.training_type_plugin.config['fp16']['min_loss_scale'] == 10
+            raise SystemExit()
+
+    model = TestModel()
+    trainer = Trainer(
+        plugins=[
+            DeepSpeedPlugin(
+                loss_scale=10, initial_scale_power=10, loss_scale_window=10, hysteresis=10, min_loss_scale=10
```
Comment (SeanNaren) [inline tests/plugins/test_deepspeed_plugin.py:319]: If anyone can suggest how to use parametize for this please do :P

Comment (carmocca) [reply tests/plugins/test_deepspeed_plugin.py:319]: For which tests? This one and `test_deepspeed_assert_config_zero_offload_disabled`? probably not worth it

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 2]
```diff
@@ -127,6 +132,18 @@ def __init__(
 
             logging_level: Set logging level for deepspeed. (Default: ``logging.WARN``)
 
+            loss_scale: Loss scaling value for FP16 training.
+                0.0 results in dynamic loss scaling, otherwise static (Default: 0)
+
+            initial_scale_power: Power of the initial dynamic loss scale value. Loss scale is computed
+                by 2^initial_scale_power (Default 32)
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/deepspeed.py:139]:
```suggestion
                by ``2^initial_scale_power`` (Default: 32)
```

**Review by carmocca — APPROVED** [review]

File: tests/plugins/test_deepspeed_plugin.py (hunk 1 above)
Comment (SeanNaren) [reply tests/plugins/test_deepspeed_plugin.py:319]: Cool!

**Review by kaushikb11 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
